src/sctools/groups.py

```diff
@@ -0,0 +1,151 @@
+            rows = contents
+        metrics[cell_id].update({
+                k: rows[k] for k in rows if k not in
+                ['SAMPLE', 'LIBRARY', 'READ_GROUP', 'CATEGORY']
+                })
+        df = pd.DataFrame.from_dict(metrics, orient='columns')
+        df.insert(0, 'Class', class_name)
+        d = d.append(df)
+    d.to_csv(output_name + '.csv')
+
+
+def aggregate_picard_metrics_by_table(file_names, output_name):
+    """
+    pipeline output picard QC metrics at single cell/sample level.
+    This function is called to merge/aggregate QC metrics by metrics type and
+    then merge multiple QC measurement into single matrix file.
+    In this file, column is sample/cell and row is QC metrics
+    :param file_names: list of files generated by Picard, metrics are in table.
+    :param output_name: output name without suffix'
+   """
+    for file_name in file_names:
+        cell_id = os.path.basename(file_name).split('_qc')[0]
+        class_name = os.path.basename(file_name).split('.')[1]
+        parsed = picard.parse(file_name)
+        dat = pd.DataFrame.from_dict(parsed['metrics']['contents'])
+        dat.insert(0, 'Sample', cell_id)
+        dat.to_csv(output_name + "_" + class_name + '.csv', index=False)
+
+
+def aggregate_qc_metrics(file_names, output_name):
+    """
+    Append several metrics into one file. 
+    :param file_names: list of files, such as Picard row metric, hisat2 metric
+    and rsem metric
+    :param output_name: output file name without suffix.
+    """
+    df = pd.DataFrame()
+    for file_name in file_names:
+        dat = pd.read_csv(file_name)
+        df = df.append(dat, ignore_index=True)
+    df.rename(columns={'Unnamed: 0': 'Metrics'}, inplace=True)
+    df.to_csv(output_name + '.csv', index=False)
+
+
+def parse_hisat2_log(file_names, output_name):
+    """
+    to parse the hisat2 log file, which include alignment summary
+    :param file_names: hisat2 log files
+    :param output_name: output file name without suffix
+    """
+    metrics = {}
+    tag = "NONE"
+    for file_name in file_names:
+        if '_qc' in file_name:
+            cell_id = os.path.basename(file_name).split('_qc')[0]
+            tag = "HISAT2G"
+        elif '_rsem' in file_name:
+            cell_id = os.path.basename(file_name).split('_rsem')[0]
+            tag = "HISAT2T"
+        with open(file_name) as f:
+            dat = f.readlines()
+            d = [x.strip().split(':') for x in dat]
+            d.pop(0)
+            metrics[cell_id] = {x[0]: x[1].strip().split(' ')[0] for x in d}
+    df = pd.DataFrame.from_dict(metrics, orient='columns')
+    df.insert(0, "Class", tag)
+    df.to_csv(output_name + '.csv')
+
+
+def parse_rsem_cnt(file_names, output_name):
+    """
+    to parse the rsem cnt file, which include alignment summary
+    :param file_names: rsem cnt files
+    :param output_name: output file name without suffix
+    """
+    metrics = {}
```

Comment:
more or less same comments here about explaining a bit more about what is going on.

Reply:
Functional-wise, it does the same thing as another function. I added a few words to describe more details